### ISSUE TITLE
if the guitar tablature has a chord with a note on each string, OSMD won't display tab 

### ISSUE THREAD

**Issue by momolarson** [opensheetmusicdisplay/opensheetmusicdisplay#911]
I created a simple tab in Muse Score and have attached it.  If you drop this file into the demo app, you get a lot of errors talking about svg heights being NaN and Infinity.  

I have included two examples in the zip:  

1> Full_Chord_Error.musicXML has the error
2> OLTB.xml - same chord, nothing on 6th string, works fine.

[chordError.zip](https://github.com/opensheetmusicdisplay/opensheetmusicdisplay/files/5462504/chordError.zip)

**Comment by momolarson** [comment]
@sschmidTU - I have been looking into this and I am willing to fix, I just wanted to get this documented!  Also, maybe some sage guidance!  The error happens here:

https://github.com/opensheetmusicdisplay/opensheetmusicdisplay/blob/6b10b01c0a5cdd73087b09541b1aa89b2a9d5b6e/src/MusicalScore/Graphical/VexFlow/VexFlowStaffEntry.ts#L58

but I am having issues determining where those values are calculated.

**Comment by momolarson** [comment]
After a little more digging, It looks like the error is occurring in BoundingBox: https://github.com/opensheetmusicdisplay/opensheetmusicdisplay/blob/6b10b01c0a5cdd73087b09541b1aa89b2a9d5b6e/src/MusicalScore/Graphical/BoundingBox.ts#L309

In the "final" call to calculateBoundingBox, the parent Bounding Box is getting the default values and then when looping through the children, the max function is called creating numbers like "5e-324" (Number.MAX_VALUE)

should these default to max width of the page? Thanks for any insight,

Gregg

**Comment by sschmidTU** [comment]
Ah, interesting! Yes, we should probably initialize with max width(/height?) of the page instead of Number.MAX_VALUE.
But why is Number.MAX_VALUE `5e-324`? In my Firefox and Chrome it's `1.7976931348623157e+308`. The exponent should never be negative.
I'll see if initializing with max page width fixes it (though a bit later, if you could test and submit a PR it would be great too!)

**Comment by momolarson** [comment]
Thanks @sschmidTU  - I appreciate that tip, and I will investigate why I am seeing such an odd MAX_VALUE, I am using Microsoft Edge (chrome), so perhaps it is something there.  I can test out the fix you mentioned too!  Hope all is well!

Gregg

**Comment by sschmidTU** [comment]
Just checked Edge, the result of MAX_VALUE is the same.
Maybe we add something to the bounding box after initializing, and then it overflows from max value to a negative value?

**Comment by sschmidTU** [comment]
Though actually `Number.MAX_VALUE + 1` or `+ 1000` is the same as `MAX_VALUE`, and `MAX_VALUE * 2` is Infinity (also in Edge, which like Chrome is based on the chromium engine).

**Comment by momolarson** [comment]
I was able to track down more specifically where and why the issue is happening, and I was able to "fix" it, but lets see what you think! 

The error originates in the "CalculateLines" function in theSkyBottomLineCalculator.ts:

https://github.com/opensheetmusicdisplay/opensheetmusicdisplay/blob/3f367bb80fb8cbc3100408581b3cccc563601a9b/src/MusicalScore/Graphical/SkyBottomLineCalculator.ts#L99

What is happening is that an alpha of 0 is being created (x val 69,70) so the y value does not get added.  this creates "gaps" in the array ID's, which gets translated into an infinity later.  This is only happening in this tab, in the cases that the tab works, there are no alphas <= 0, and the y value is usually 164.

Here is a screenshot showing part of tmpSkyLine before it is pushed into mSkyLine

![gaps](https://user-images.githubusercontent.com/43911543/97781142-198d7780-1b4f-11eb-8194-fd496d00948c.JPG)

https://github.com/opensheetmusicdisplay/opensheetmusicdisplay/blob/3f367bb80fb8cbc3100408581b3cccc563601a9b/src/MusicalScore/Graphical/SkyBottomLineCalculator.ts#L105

Once the Push is complete, and afer the image is drawn (lines 111-114) mSkyLine looks like:  

![gapundefined](https://user-images.githubusercontent.com/43911543/97781267-bc45f600-1b4f-11eb-974d-cb373868a6c1.JPG)

My "hack fix" was to make sure that tmlSkyLine always had an x value with:

```
                if(tmpSkyLine[x] === undefined) {
                    tmpSkyLine[x] = 0; //tried 164 and it works too
                }
```

But I am not sure that is acceptable and perhaps you would have a better idea for me to explore?

**Comment by sschmidTU** [comment]
Ah, that makes sense, great that you found a (quick) fix!
It would probably be better to make the undefined values average between the previous and next values though,
because the skyline is supposed to represent the highest y value in the score at that position.
i think it will be easy to make such a fix.

**Comment by momolarson** [comment]
Cool, yeah, I can get that done

**Comment by sschmidTU** [comment]
Fixed in #912 by @momolarson - thank you very much!

**Comment by sschmidTU** [comment]
![image](https://user-images.githubusercontent.com/33069673/97786043-0dce9f00-1ba9-11eb-9ebf-e3d9909c66ae.png)
